lib/rules/check-ignored-types.js

```diff
@@ -0,0 +1,41 @@
+"use strict";
+
+var ignoredBaseTypes = ["ExperimentalSpreadProperty"];
+var ignoredKeyTypes = ["CallExpression", "ConditionalExpression"];
```

Comment:
I think it will simplify the code in `checkIgnoredType` if all `ignore*Types` arrays share the same structure, so these could be arrays of objects with the `type` key.